src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -396,7 +396,8 @@ private static boolean enumEqCheck(Object value, EnumDesc<?> label, MethodHandle
             try {
                 Class<?> clazz = label.constantType().resolveConstantDesc(lookup);
 
-                if (value.getClass() != clazz) {
+                if (!(value instanceof Enum<?> enumValue) ||
```

Comment:
Minor comment: If we are more concerned about performance when the enum is in a mix-and-match with other types (say primitives etc.), we can move the `instanceof` test before the `constantType().resolveConstantDesc()` to avoid a Class lookup.